xcp_d/utils/hcp2fmriprep.py

```diff
@@ -233,101 +303,41 @@ def hcpfmriprepx(hcp_dir, out_dir, sub_id):
-        ciftip = k + "/" + filenamex + "_Atlas_MSMAll.dtseries.nii"
-        ciftib = (
-            funcdir
-            + "/sub-"
-            + sub_id
-            + "_task-"
-            + idx[1]
-            + "_acq-"
-            + idx[2]
-            + "_space-fsLR_den-91k_bold.dtseries.nii"
-        )
-
-        niftip = k + "/" + filenamex + ".nii.gz"
-        TR = nb.load(niftip).header.get_zooms()[-1]  # repetition time
-
-        jsontis = {"RepetitionTime": np.float(TR), "TaskName": idx[1]}
-
-        json2 = {
-            "grayordinates": "91k",
-            "space": "HCP grayordinates",
-            "surface": "fsLR",
-            "surface_density": "32k",
-            "volume": "MNI152NLin6Asym",
-        }
-
-        boldjson = (
-            funcdir
-            + "/sub-"
-            + sub_id
-            + "_task-"
-            + idx[1]
-            + "_acq-"
-            + idx[2]
-            + "_space-MNI152NLin6Asym_desc-preproc_bold.json"
-        )
-        ddjsonfile = out_dir + "/dataset_description.json"
-        ciftijson = (
-            funcdir
-            + "/sub-"
-            + sub_id
-            + "_task-"
-            + idx[1]
-            + "_acq-"
-            + idx[2]
-            + "_space-fsLR_den-91k_bold.dtseries.json"
-        )
-
-        ddjson = {"Name": "HCP", "DatasetType": "derivative", "GeneratedBy": [{"Name": "HCP"}]}
-
-        writejson(jsontis, boldjson)
-        writejson(json2, ciftijson)
-        writejson(ddjson, ddjsonfile)
-
-        # get the files
-        rawfiles = [hcp_ref, ciftip]
-        newfiles = [prep_ref, ciftib]
-
-        for kk, jj in zip(rawfiles, newfiles):
-            copyfileobj_example(kk, jj)
-            print("finished converting" + kk)
+        regressors.to_csv(regressors_tsv_fmriprep, index=False, sep="\t")
+
+        # NOTE: Is this JSON any good?
+        regressors_json_fmriprep = os.path.join(func_dir_fmriprep, f"{regressors_file_base}.json")
+        regressors.to_json(regressors_json_fmriprep)
+
+    # Copy HCP files to fMRIPrep folder
+    for file_orig, files_fmriprep in copy_dictionary.items():
+        if not isinstance(files_fmriprep):
+            raise ValueError(
+                f"Entry for {file_orig} should be a list, but is a {type(files_fmriprep)}"
+            )
+
+        for file_fmriprep in files_fmriprep:
+            copyfileobj_example(file_orig, file_fmriprep)
+
+    # Write the dataset description out last
+    # TODO: Add "unknown" version to dictionary.
+    dataset_description_dict = {
+        "Name": "HCP",
+        "DatasetType": "derivative",
+        "GeneratedBy": [
+            {"Name": "HCP"},
+        ],
+    }
+    dataset_description_fmriprep = os.path.join(out_dir, "dataset_description.json")
+    writejson(dataset_description_dict, dataset_description_fmriprep)
+
+    # Write out the mapping from HCP to fMRIPrep
+    mapping_fmriprep = os.path.join(subject_dir_fmriprep, "file_mapping.json")
```

Comment:
I need a more BIDS-y name for this. What about using the "scans.tsv" file?

Reply:
Also, there was a BEP about file mapping at some point. I think it was turned into a tool instead of added to BIDS.

Reply:
The tool doesn't output a manifest, so I think I'll just go with the scans.tsv file.

Reply:
It also could be useful for provenance (#568).

Reply:
I ended up using scans.tsv.